doc/api/worker_threads.md

```diff
@@ -396,6 +396,74 @@ posting without having side effects.
 For more information on the serialization and deserialization mechanisms
 behind this API, see the [serialization API of the `v8` module][v8.serdes].
 
+#### Considerations when transferring TypedArrays and Buffers
+
+All `TypedArray` and `Buffer` instances are views over an underlying
+`ArrayBuffer`. That is, it is the `ArrayBuffer` that actually stores
+the raw data while the `TypedArray` and `Buffer` objects provide a
+way of viewing and manipulating the data. It is possible and common
+for multiple views to be created over the same `ArrayBuffer` instance.
+
+For example,
+
+```js
+const ab = new ArrayBuffer(10);
+
+// Both TypedArrays share the same underlying state.
+const u1 = new Uint8Array(ab);
+const u2 = new Uint16Array(ab);
+
+// Changing one, changes the other
+u1[1] = 1;
+
+console.log(u2[0]);  // prints 256
+```
+
+Great care must be taken when using a transfer list to transfer an
+`ArrayBuffer` as doing so may cause all `TypedArray` and `Buffer`
+instances that share that same `ArrayBuffer` to become unusable.
+
+```js
+const ab = new ArrayBuffer(10);
+
+const u1 = new Uint8Array(ab);
+const u2 = new Uint16Array(ab);
+
+port.postMessage(u1, [u1.buffer]);
+
+console.log(u2.length);  // prints 0
+```
+
+For `Buffer` instances, specifically, whether the `Buffer` can be
+safely transferred depends entirely on how instances were created,
+which often cannot be reliably determined.
+
+Node.js will create `Buffer` instances with data backed by either
+a shared pre-allocated pool (that is, a common `ArrayBuffer`), or
+with their own dedicated `ArrayBuffer` instance. In addition to
+using the JavaScript `Buffer.from()`, `Buffer.alloc()`, and
+`Buffer.allocUnsafe()` methods to create `Buffer` instances, Node.js
+uses other mechanisms at the native C++ level to create `Buffer`
+instances over memory regions that may be allocated and accessed
+independently of the `Buffer`. The point is, whenever JavaScript
+code is using a `Buffer` instance, the exact ownership over the
+underlying data, along with whatever other code may be using or
+relying on that data, may not be completely clear. Transferring
+such objects can be dangerous and can lead to undefined behavior
+or even crash the Node.js process.
+
+Further, `Buffer` instances created using either `Buffer.from()` or
+`Buffer.allocUnsafe()` will *always* be cloned and will never be
+transferred. However, because those are created from the
+underlying shared pre-allocation pool, sending these `Buffer`
+instances will cause the *entire* pre-allocation pool to also
+be cloned, potentially leading to higher unintended memory use,
+memory leaks and possible security issues.
+
+Because of these issues, unless a `Buffer` instance is definitely
+known to be safe, it is best to avoid transferring `Buffer` objects
+at all.
+
```

Comment:
```suggestion
Depending on how a `Buffer` instance was created, it may or may not
own its underlying `ArrayBuffer`. An `ArrayBuffer` must not be
transferred unless it is known that the `Buffer` instance owns its underlying
`ArrayBuffer`. In particular, for `Buffer`s created from the internal `Buffer`
pool, transferring them is not possible for that reason, and they will
always be cloned, which includes sending a copy of the entire `Buffer`
pool with them. This behavior may come with unintended higher
memory usage.

See [`Buffer.allocUnsafe()`][] for more details on `Buffer` pooling.

The `ArrayBuffer`s for `Buffer`s instances created using
`Buffer.alloc()` or `Buffer.allocUnsafeSlow()` can always safely be
transferred if no other views of those `ArrayBuffer`s have been created.
```